physicsnemo/utils/neighbors/knn/_torch_impl.py

```diff
@@ -0,0 +1,45 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+
+def knn_impl(
+    points: torch.Tensor,
+    queries: torch.Tensor,
+    k: int = 3,
+) -> torch.Tensor:
+    """
+    Perform kNN search with torch.
+
+    Args:
+        points (torch.Tensor): Query points, shape ( M, D)
+        queries (torch.Tensor): Reference points, shape (N, D)
+        k (int): Number of neighbors
+
+    Returns:
+        torch.Tensor: Aggregated features at p1, shape (N, k)
```

Comment:
I thought I pointed this out in the last round of reviews, but apparently not?

This docstring return signature doesn't seem correct